Add user-agent string to Elasticsearch templates

vidok opened the pull request:
Our team at Elastic is interested to have metrics to understand how popular the templates are. With that data we might justify investing to help to maintain them. 

This PR just adds a user-agent string to BigQuery/GCS/PubSub to Elasticsearch templates.

The changes are tested in GCP

vidok commented:
@shreyakhajanchi @fbiville @liferoad Sorry for writing you directly folks. I just wonder if there is anything else I need to do to make this PR merged. Thanks in advance.

liferoad commented:
@vidok thanks a lot for this work. Can you get LGTM from your team at Elastic first?

vidok commented:
> @vidok thanks a lot for this work. Can you get LGTM from your team at Elastic first? Please also fix the failed tests.

@liferoad I've fixed the failed specs and an extra approval from our team is coming :). Thank you for looking at it.

seanstory commented on v2/elasticsearch-common/src/main/java/com/google/cloud/teleport/v2/elasticsearch/utils/ElasticsearchIO.java:
```diff
@@ -414,6 +422,29 @@ public ConnectionConfiguration withTrustSelfSignedCerts(boolean trustSelfSignedC
+    protected Header[] configureDefaultHeaders() {
+      final ArrayList<Header> headerList = new ArrayList<Header>();
+
+      headerList.add(new BasicHeader("user-agent", getUserAgent()));
```
```suggestion
      headerList.add(new BasicHeader("User-Agent", getUserAgent()));
```
Nit. Casing _shouldn't_ matter, but I believe this shouldn't be all lowercase.

vidok replied:
HTTP headers are case-insensitive but I can change it because capitalised name looks better :).

seanstory reviewed (approved):
LGTM. Left 1 nit

liferoad commented:
> > @vidok thanks a lot for this work. Can you get LGTM from your team at Elastic first? Please also fix the failed tests.
> 
> @liferoad I've fixed the failed specs and an extra approval from our team is coming :). Thank you for looking at it.

codecov seems still failing.

vidok commented:
> > > @vidok thanks a lot for this work. Can you get LGTM from your team at Elastic first? Please also fix the failed tests.
> > 
> > 
> > @liferoad I've fixed the failed specs and an extra approval from our team is coming :). Thank you for looking at it.
> 
> codecov seems still failing.

Ah, I didn't get it. Is this because of the code coverage decrease?

liferoad commented:
> > > > @vidok thanks a lot for this work. Can you get LGTM from your team at Elastic first? Please also fix the failed tests.
> > > 
> > > 
> > > @liferoad I've fixed the failed specs and an extra approval from our team is coming :). Thank you for looking at it.
> > 
> > 
> > codecov seems still failing.
> 
> Ah, I didn't get it. Is this because of the code coverage decrease?

Yes. some minor decrease.

https://app.codecov.io/gh/GoogleCloudPlatform/DataflowTemplates/pull/1393?dropdown=coverage&src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=checks&utm_campaign=pr+comments&utm_term=GoogleCloudPlatform

vidok commented:
@liferoad I pushed some unit tests. I'd appreciate if you can have a look.

liferoad commented:
> @liferoad I pushed some unit tests. I'd appreciate if you can have a look.

I rebased your PR and please check the failed checks.

vidok commented:
> > @liferoad I pushed some unit tests. I'd appreciate if you can have a look.
> 
> I rebased your PR and please check the failed checks.

Thanks! There was some styling issues and I fixed them.

liferoad commented:
codecov still failed.

liferoad reviewed (approved):
The failed tests are not related to this PR. Approve this.
